.github/workflows/benchmark_on_push.yml

```
@@ -3,6 +3,11 @@ on:
   push:
     branches: [main, develop]
 
+ 
+permissions:
+  contents: read
+  pull-requests: read
```

```suggestion
permissions: {}
```

and then we can set both of these in the context of the `benchmarks:` job below (like you've done for other files).